src/GraphQL/Utilities/SchemaBuilder.cs

```diff
@@ -381,6 +392,8 @@ protected virtual EnumerationGraphType ToEnumerationType(GraphQLEnumTypeDefiniti
                 Name = enumDef.Name.Value,
                 Description = typeConfig.Description ?? enumDef.Comment?.Text
             };
+            type.SetAstType(enumDef);
```

Comment:
So `SetAstType` has become much used I suggest to change
```c#
public static void SetAstType<T>(this IProvideMetadata type, T node) where T : ASTNode
{
    type.Metadata[__AST_MetaField__] = node;
}
```
to
```c#
public static TMetadataProvider SetAstType<TMetadataProvider>(this TMetadataProvider type, ASTNode node)
    where TMetadataProvider : IProvideMetadata => type.WithMetadata(__AST_MetaField__, node);
```
to be able to do this:
```c#
var type = new EnumerationGraphType
{
    Name = enumDef.Name.Value,
    Description = typeConfig.Description ?? enumDef.Comment?.Text
}.SetAstType(enumDef);
```

Reply:
Done

Reply:
Done